packages/material-ui/src/Container/Container.d.ts

```diff
@@ -3,6 +3,7 @@ import { StandardProps } from '..';
 
 export interface ContainerProps
   extends StandardProps<React.HTMLAttributes<HTMLDivElement>, ContainerClassKey> {
+  component?: React.ElementType<React.HTMLAttributes<HTMLElement>>;
```

Comment:
To be confirmed with @eps1lon

```suggestion
  component?: React.ElementType<React.HTMLAttributes<HTMLDivElement>>;
```